plugins/inputs/intel_powerstat/intel_powerstat.go

```diff
@@ -0,0 +1,493 @@
+		"name":  "current_dram_power_consumption",
+		"unit":  "Watt",
+		"value": services.RoundFloatToNearestTwoDecimalPlaces(p.rapl.GetCurrentDramPowerConsumption(socketID)),
+	}
+
+	acc.AddGauge("powerstat_package", fields, tags)
+}
+
+func (p *PowerStat) verifyProcessor() error {
+	stats, err := p.fs.GetCPUInfoStats()
+	p.cpuInfo = stats
+
+	if err != nil {
+		return err
+	}
+
+	// First CPU is sufficient for verification.
+	firstCPU := p.cpuInfo["0"]
+	if firstCPU == nil {
+		return fmt.Errorf("first core not found while parsing /proc/cpuinfo")
+	}
+
+	if firstCPU.VendorID != "GenuineIntel" || firstCPU.CPUFamily != "6" {
+		return fmt.Errorf("Intel processor not found, vendorId: %s", firstCPU.VendorID)
+	}
+
+	// Disable c1CoreStateResidency and c6CoreStateResidency for specific processors.
+	allowedProcessors, err := services.ConvertHexArrayToIntegerArray(allowedProcessorModelsForC1C6)
+	if err != nil {
+		return err
+	}
+	if !contains(services.ConvertIntegerArrayToStringArray(allowedProcessors), firstCPU.Model) {
+		p.c1CoreStateResidency = false
+		p.c6CoreStateResidency = false
+	}
+
+	if !strings.Contains(firstCPU.Flags, "msr") {
+		p.coreCPUTemperature = false
+		p.c6CoreStateResidency = false
+		p.processorBusyCycles = false
+		p.processorBusyFrequency = false
+		p.c1CoreStateResidency = false
+	}
+
+	if !strings.Contains(firstCPU.Flags, "aperfmperf") {
+		p.processorBusyFrequency = false
+		p.processorBusyCycles = false
+		p.c1CoreStateResidency = false
+	}
+
+	if !strings.Contains(firstCPU.Flags, "dts") {
+		p.coreCPUTemperature = false
+	}
+
+	return nil
+}
+
+// ParseCPUMetricsConfig enables specific metrics based on config values.
+func (p *PowerStat) ParseCPUMetricsConfig() {
+	if len(p.CPUMetrics) == 0 {
+		p.perCoreMetrics = false
+		return
+	}
+
+	if contains(p.CPUMetrics, cpuFrequency) {
+		p.coreCPUFrequency = true
+	}
+
+	if contains(p.CPUMetrics, cpuC1) {
+		p.c1CoreStateResidency = true
+	}
+
+	if contains(p.CPUMetrics, cpuC6) {
+		p.c6CoreStateResidency = true
+	}
+
+	if contains(p.CPUMetrics, cpuBusyCycles) {
+		p.processorBusyCycles = true
+	}
+
+	if contains(p.CPUMetrics, cpuBusyFrequency) {
+		p.processorBusyFrequency = true
+	}
+
+	if contains(p.CPUMetrics, cpuTemperature) {
+		p.coreCPUTemperature = true
+	}
+}
+
+func contains(slice []string, str string) bool {
+	for _, v := range slice {
+		if v == str {
+			return true
+		}
+	}
+
+	return false
+}
+
+// NewPowerStat creates and returns PowerStat struct.
+func NewPowerStat(fs services.FileService) *PowerStat {
+	p := &PowerStat{
+		perCoreMetrics:         false,
+		coreCPUFrequency:       false,
+		c1CoreStateResidency:   false,
+		c6CoreStateResidency:   false,
+		processorBusyCycles:    false,
+		coreCPUTemperature:     false,
+		processorBusyFrequency: false,
+		skipFirstIteration:     true,
+		fs:                     fs,
+	}
+
+	return p
+}
+
+func init() {
+	inputs.Add("intel_powerstat", func() telegraf.Input {
+		return NewPowerStat(services.NewFileService())
```

Comment:
Can't you just do
```suggestion
		return &PowerStat{}
```
and set `fs` in `Init()`?

Reply:
If `fs` is set in `Init()` it would be very difficult to test it. So it was done on purpose.

Reply:
Would like to leave it as it is right now.